ddtrace/tracer/context.go

```diff
@@ -19,10 +21,16 @@ func SpanFromContext(ctx context.Context) (Span, bool) {
 		return &ddtrace.NoopSpan{}, false
 	}
 	s := opentracing.SpanFromContext(ctx)
-	if s != nil {
-		return s.(ddtrace.Span), true
+	span, ok := s.(ddtrace.Span)
+	if !ok {
+		// span on context is not nil but it's not ddtrace.Span
+		// explicitly log unsupported behavior
+		if s != nil {
+			log.Println(warningPrefix + "span on context was not produced by ddtrace, ignoring it")
```

Comment:
`found invalid span. Only spans generated by the in-built tracer are valid. Third-party traces are not supported.`